src/messaging/tests/TestMessageCounterSyncMgr.cpp

```diff
@@ -71,11 +71,7 @@ class LoopbackTransport : public Transport::Base
 
     CHIP_ERROR SendMessage(const PacketHeader & header, const PeerAddress & address, System::PacketBufferHandle msgBuf) override
     {
-        // We need to change the peerKeyId to local Key Id to be received by itself.
-        PacketHeader tmpHeader = header;
-
-        tmpHeader.SetEncryptionKeyID(kTestLocalGroupKeyId);
-        HandleMessageReceived(tmpHeader, address, std::move(msgBuf));
+        HandleMessageReceived(header, address, std::move(msgBuf));
```

Comment:
But shouldn't the sender now be setting the remote key id. so we still need this temp header, but now we'd be swizzling the key from remote to local?

Reply:
The underlying layer (`SecureMessageCodec`) set the key ID correctly to remote's key ID. So, it will already be correctly set by the time `LoopbackTransport::SendMessage` is called.